src/main/scala/io/shiftleft/fuzzyc2cpg/passes/AstCreationPass.scala

```diff
@@ -0,0 +1,71 @@
+package io.shiftleft.fuzzyc2cpg.passes
+
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.{File, NamespaceBlock}
+import io.shiftleft.fuzzyc2cpg.passes.astcreation.{AntlrCModuleParserDriver, AstVisitor}
+import io.shiftleft.fuzzyc2cpg.Global
+import io.shiftleft.passes.{DiffGraph, KeyPool, ParallelCpgPass}
+import io.shiftleft.semanticcpg.language._
+import org.slf4j.LoggerFactory
+
+/**
+  * Given a list of filenames, this pass creates abstract syntax trees for
+  * each file. Files are processed in parallel.
+  * */
+class AstCreationPass(filenames: List[String], cpg: Cpg, keyPools: Option[Iterator[KeyPool]])
+    extends ParallelCpgPass[String](cpg, keyPools = keyPools) {
+
+  private val logger = LoggerFactory.getLogger(getClass)
+  val global: Global = Global()
+
+  override def partIterator: Iterator[String] = filenames.iterator
+
+  override def runOnPart(filename: String): Iterator[DiffGraph] = {
+    fileAndNamespaceBlock(filename) match {
+      case Some((fileNode, namespaceBlock)) =>
+        val driver = createDriver(fileNode, namespaceBlock)
+        tryToParse(driver, filename)
+      case None =>
+        logger.warn("Invalid File/Namespace Graph")
+        Iterator()
+    }
+  }
+
+  private def fileAndNamespaceBlock(filename: String): Option[(File, NamespaceBlock)] = {
+    val absolutePath = new java.io.File(filename).getAbsolutePath
+    cpg.file
+      .nameExact(absolutePath)
+      .l
+      .flatMap { f =>
+        f.start.namespaceBlock.l.map(n => (f, n))
+      }
+      .headOption
+  }
```

Comment:
Well, what you're proposing is essentially to merge `FileAndNamespaceBlockPass` and `AstCreatorPass`, because passes can't pass NewNode instances from one another. That would imply that these nodes are not yet in the graph, and hence, they'd need to be written by the receiving pass. Overall, this breaks the model where a pass receives a graph and some auxiliary information and then applies diffs to that graph. We need to lookup file nodes anyway, and the namespaces are just an edge away, so, I'd be surprised if this causes performance trouble. Consider also that it means that the user of the pass needs to understand our file/namespaceblock association, which in its current state is still simple but would be more complex if we actually handled namespaces. 

I'll try merging the two, that should be better.